### PR TITLE
[GHA] Check OpImplCheck conformance for template plugin (to make sure that all functions are checked by conformance)

### PR DESCRIPTION
**Pull request by iefode**
### Details:
 - *Fix win run*

### Tickets:
 - *[ticket-id](https://jira.devtools.intel.com/browse/CVS-123572)*

### CODE REVIEW

File: .github/workflows/windows.yml [hunk 1]
```diff
@@ -571,13 +571,7 @@ jobs:
...
         shell: cmd
         run: |
-          call "${{ env.INSTALL_DIR }}\\setupvars.bat" && ${{ env.INSTALL_TEST_DIR }}/subgraphsDumperTests --gtest_print_time=1  --gtest_print_time=1 --device=TEMPLATE --gtest_filter="*OpImpl*" --gtest_output=xml:${{ env.INSTALL_TEST_DIR }}/TEST-SubgraphsDumperTests.xml
-
-      - name: Template OpImpl tests
-        if: ${{ 'false' }} # Ticket: 123572
-        shell: cmd
-        run: |
-          call "${{ env.INSTALL_DIR }}\\setupvars.bat" && ${{ env.INSTALL_TEST_DIR }}/conformanceTests --gtest_output=xml:${{ env.INSTALL_TEST_DIR }}/TEST-TemplateOpImplTests.xml
+          call "${{ env.INSTALL_DIR }}\\setupvars.bat" && ${{ env.INSTALL_TEST_DIR }}/subgraphsDumperTests --gtest_print_time=1  --gtest_print_time=1 --gtest_output=xml:${{ env.INSTALL_TEST_DIR }}/TEST-SubgraphsDumperTests.xml
```
Comment (mryzhov) [inline .github/workflows/windows.yml:574]:
```suggestion
          call "${{ env.INSTALL_DIR }}\\setupvars.bat" && ${{ env.INSTALL_TEST_DIR }}/subgraphsDumperTests --gtest_print_time=1  --gtest_output=xml:${{ env.INSTALL_TEST_DIR }}/TEST-SubgraphsDumperTests.xml
```

**Comment by ilya-lavrenov** [comment]
Invalid configuration and CI is not run https://github.com/openvinotoolkit/openvino/actions/runs/6654482062

File: .github/workflows/linux.yml [hunk 2]
```diff
@@ -416,17 +416,16 @@ jobs:
       max-parallel: 2
       fail-fast: false
       matrix:
-        include:
-          # 'OP' for Opset, 'API' for API
-          - TEST_TYPE: 'OP'
-          - TEST_TYPE: 'API'
+        # 'OP' for Opset, 'API' for API
+        - TEST_TYPE: ['OP', 'API']
```
Comment (mryzhov) [inline .github/workflows/linux.yml:420]:
```suggestion
        TEST_TYPE: [OP, API]
```

File: .github/workflows/linux.yml [hunk 3]
```diff
@@ -416,17 +416,16 @@ jobs:
...
+        - TEST_DEVICE: ['CPU', 'TEMPLATE']
```
Comment (mryzhov) [inline .github/workflows/linux.yml:421]:
```suggestion
        TEST_DEVICE: [CPU, TEMPLATE]
```

**Review by ilya-lavrenov — CHANGES_REQUESTED** [review]
Agree to have it in precommit only iff it does not consume a lot of time.
Template plugin is not a first class citizen to waste precommit time

**Comment by iefode** [comment]
@akashchi Please take a look

**Review by akashchi — APPROVED** [review]

**Pull request merged**